File: projects/remote-api/source/java/org/alfresco/repo/webservice/classification/ClassificationWebService.java
/*
 * Copyright (C) 2005 Alfresco, Inc.
 *
 * Licensed under the Mozilla Public License version 1.1 
 * with a permitted attribution clause. You may obtain a
 * copy of the License at
 *
 *   http://www.alfresco.org/legal/license.txt
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific
 * language governing permissions and limitations under the
 * License.
 */
package org.alfresco.repo.webservice.classification;

import java.rmi.RemoteException;

import org.alfresco.repo.webservice.AbstractWebService;
import org.alfresco.repo.webservice.types.Category;
import org.alfresco.repo.webservice.types.ClassDefinition;
import org.alfresco.repo.webservice.types.Classification;
import org.alfresco.repo.webservice.types.Predicate;
import org.alfresco.repo.webservice.types.Reference;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

/**
 * Web service implementation of the ClassificationService.
 * The WSDL for this service can be accessed from http://localhost:8080/alfresco/wsdl/classification-service.wsdl
 *  
 * @author gavinc
 */
public class ClassificationWebService extends AbstractWebService implements ClassificationServiceSoapPort
{
   private static Log logger = LogFactory.getLog(ClassificationWebService.class);

   /**
    * @see org.alfresco.repo.webservice.classification.ClassificationServiceSoapPort#getClassifications()
    */
   public Classification[] getClassifications() throws RemoteException, ClassificationFault
   {
      throw new ClassificationFault(1, "getClassifications() is not implemented yet!");
   }

   /**
    * @see org.alfresco.repo.webservice.classification.ClassificationServiceSoapPort#getChildCategories(org.alfresco.repo.webservice.types.Reference)
    */
   public Category[] getChildCategories(Reference parentCategory) throws RemoteException, ClassificationFault
   {
      throw new ClassificationFault(1, "getChildCategories() is not implemented yet!");
   }

   /**
    * @see org.alfresco.repo.webservice.classification.ClassificationServiceSoapPort#getCategories(org.alfresco.repo.webservice.types.Predicate)
    */
   public CategoriesResult[] getCategories(Predicate items) throws RemoteException, ClassificationFault
   {
      throw new ClassificationFault(1, "getCategories() is not implemented yet!");
   }

   /**
    * @see org.alfresco.repo.webservice.classification.ClassificationServiceSoapPort#setCategories(org.alfresco.repo.webservice.types.Predicate, org.alfresco.repo.webservice.classification.AppliedCategory[])
    */
   public CategoriesResult[] setCategories(Predicate items, AppliedCategory[] categories) throws RemoteException, ClassificationFault
   {
      throw new ClassificationFault(1, "setCategories() is not implemented yet!");
   }

   /**
    * @see org.alfresco.repo.webservice.classification.ClassificationServiceSoapPort#describeClassification(org.alfresco.repo.webservice.types.Reference)
    */
   public ClassDefinition describeClassification(Reference classification) throws RemoteException, ClassificationFault
   {
      throw new ClassificationFault(1, "describeClassification() is not implemented yet!");
   }
}

File: projects/extensions/php/alfresco/webservice/ClassificationWebService.php
<?php

/*
  Copyright (C) 2005 Alfresco, Inc.

  Licensed under the Mozilla Public License version 1.1
  with a permitted attribution clause. You may obtain a
  copy of the License at

    http://www.alfresco.org/legal/license.txt

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
  either express or implied. See the License for the specific
  language governing permissions and limitations under the
  License.
*/

require_once 'SOAP/Client.php';
require_once 'alfresco/webservice/WebServiceUtils.php';

class ClassificationWebService extends SOAP_Client
{
    function WebService_ClassificationService_ClassificationService($path = null)
    {
        if ($path == null)
        {
            $path = getServerLocation().'/alfresco/api/ClassificationService';
        }
        $this->SOAP_Client($path, 0);
    }
    function &getClassifications($store)
    {
        $getClassifications =& new SOAP_Value('{http://www.alfresco.org/ws/service/classification/1.0}getClassifications', false, $v = array('store' => $store));
        return $this->call('getClassifications',
                           $v = array('getClassifications' => $getClassifications),
                           array('namespace' => 'http://www.alfresco.org/ws/service/classification/1.0',
                                 'soapaction' => 'http://www.alfresco.org/ws/service/classification/1.0/getClassifications',
                                 'style' => 'document',
                                 'use' => 'literal'));
    }
    function &getChildCategories($parentCategory)
    {
        $getChildCategories =& new SOAP_Value('{http://www.alfresco.org/ws/service/classification/1.0}getChildCategories', false, $v = array('parentCategory' => $parentCategory));
        return $this->call('getChildCategories',
                           $v = array('getChildCategories' => $getChildCategories),
                           array('namespace' => 'http://www.alfresco.org/ws/service/classification/1.0',
                                 'soapaction' => 'http://www.alfresco.org/ws/service/classification/1.0/getChildCategories',
                                 'style' => 'document',
                                 'use' => 'literal'));
    }
    function &getCategories($items)
    {
        $getCategories =& new SOAP_Value('{http://www.alfresco.org/ws/service/classification/1.0}getCategories', false, $v = array('items' => $items));
        return $this->call('getCategories',
                           $v = array('getCategories' => $getCategories),
                           array('namespace' => 'http://www.alfresco.org/ws/service/classification/1.0',
                                 'soapaction' => 'http://www.alfresco.org/ws/service/classification/1.0/getCategories',
                                 'style' => 'document',
                                 'use' => 'literal'));
    }
    function &setCategories($items, $categories)
    {
        $setCategories =& new SOAP_Value('{http://www.alfresco.org/ws/service/classification/1.0}setCategories', false, $v = array('items' => $items, 'categories' => $categories));
        return $this->call('setCategories',
                           $v = array('setCategories' => $setCategories),
                           array('namespace' => 'http://www.alfresco.org/ws/service/classification/1.0',
                                 'soapaction' => 'http://www.alfresco.org/ws/service/classification/1.0/setCategories',
                                 'style' => 'document',
                                 'use' => 'literal'));
    }
    function &describeClassification($classification)
    {
        $describeClassification =& new SOAP_Value('{http://www.alfresco.org/ws/service/classification/1.0}describeClassification', false, $v = array('classification' => $classification));
        return $this->call('describeClassification',
                           $v = array('describeClassification' => $describeClassification),
                           array('namespace' => 'http://www.alfresco.org/ws/service/classification/1.0',
                                 'soapaction' => 'http://www.alfresco.org/ws/service/classification/1.0/describeClassification',
                                 'style' => 'document',
                                 'use' => 'literal'));
    }
}

?>
